src/pkg/cli/tail.go

```diff
@@ -298,7 +298,7 @@ func tail(ctx context.Context, provider client.Provider, projectName string, opt
 					spaces, _ = term.Warnf("Reconnecting...\r") // overwritten below
 				}
 				pkg.SleepWithContext(ctx, 1*time.Second)
-				serverStream, err = provider.Follow(ctx, &defangv1.TailRequest{Services: options.Services, Etag: options.Etag, Since: timestamppb.New(options.Since)})
+				serverStream, err = provider.Follow(ctx, &defangv1.TailRequest{Project: projectName, Services: options.Services, Etag: options.Etag, Since: timestamppb.New(options.Since), LogType: uint32(options.LogType)})
```

Comment:
Wow I can't believe I didn't notice this when I added the logtype. Why don't we re-use the same tail request from the original call to `Follow`?

```diff
diff --git a/src/pkg/cli/tail.go b/src/pkg/cli/tail.go
index f8c00a38..37f420a6 100644
--- a/src/pkg/cli/tail.go
+++ b/src/pkg/cli/tail.go
@@ -213,13 +213,15 @@ func tail(ctx context.Context, provider client.Provider, projectName string, opt
                since = timestamppb.New(options.Since)
        }
 
-       serverStream, err := provider.Follow(ctx, &defangv1.TailRequest{
+       tailRequest := &defangv1.TailRequest{
                Project:  projectName,
                Services: options.Services,
                Etag:     options.Etag,
                Since:    since,
                LogType:  uint32(options.LogType),
-       })
+       }
+
+       serverStream, err := provider.Follow(ctx, tailRequest)
        if err != nil {
                return err
        }
```

If we break out `tailRequest` up above, then we can do this:

```suggestion
				serverStream, err = provider.Follow(ctx, tailRequest)
```

Do we need to update the timestamp? Will we get duplicate logs?

Reply:
I think this could almost work. We do update `options.Since` so will still need to set that, or simply keep the one in `TailRequest` updated?